Z3 proof backend bug

I think the temporary variable introduced in #331 makes the printer of the counterexample model crash:

```
dune exec catala -- Proof examples/aides_logement/aides_logement.catala_fr -O
[ERROR] [BaseMensuelleAllocationsFamiliales.montant] This variable might return an empty error:
 --> examples/aides_logement/../base_mensuelle_allocations_familiales/bmaf.catala_fr
  | 
6 |   sortie montant contenu argent
  |          ^^^^^^^
  + Montant de la base mensuelle des allocations familiales
The solver generated the following counterexample to explain the faulty behavior:
--> date_courante : 1900-01-01
catala: internal error, uncaught exception:
        Not_found
        Raised at Stdlib__Map.Make.find in file "map.ml", line 137, characters 10-25
        Called from Verification__Z3backend.print_model.(fun) in file "compiler/verification/z3backend.real.ml", line 227, characters 21-62
        Called from Stdlib__Format.pp_print_list in file "format.ml", line 1193, characters 4-14
        Called from Stdlib__Format.kasprintf.k in file "format.ml", line 1391, characters 4-22
        Called from Verification__Io.MakeBackendIO.print_negative_result in file "compiler/verification/io.ml", line 126, characters 17-42
        Called from Verification__Io.MakeBackendIO.encode_and_check_vc in file "compiler/verification/io.ml", line 156, characters 29-73
        Called from Verification__Solver.solve_vc.(fun) in file "compiler/verification/solver.ml", line 44, characters 11-55
        Called from Stdlib__List.fold_left in file "list.ml", line 121, characters 24-34
        Called from Verification__Solver.solve_vc in file "compiler/verification/solver.ml", line 42, characters 4-156
        Called from Driver.driver in file "compiler/driver.ml", line 237, characters 12-58
        Called from Cmdliner_term.app.(fun) in file "cmdliner_term.ml", line 24, characters 19-24
        Called from Cmdliner_eval.run_parser in file "cmdliner_eval.ml", line 34, characters 37-44
```

--- a/compiler/verification/z3backend.real.ml
+++ b/compiler/verification/z3backend.real.ml
@@ -212,7 +212,7 @@ let print_model (ctx : context) (model : Model.model) : string =
   let decls = Model.get_decls model in
   Format.asprintf "%a"
     (Format.pp_print_list
-       ~pp_sep:(fun fmt () -> Format.fprintf fmt "\n")
+       ~pp_sep:(fun fmt () -> Format.fprintf fmt "")
        (fun fmt d ->
          if FuncDecl.get_arity d = 0 then
            (* Constant case *)
@@ -222,13 +222,15 @@ let print_model (ctx : context) (model : Model.model) : string =
              failwith
                "[Z3 model]: A variable does not have an associated Z3 solution"
            (* Print "name : value\n" *)
-           | Some e ->
+           | Some e -> (
              let symbol_name = Symbol.to_string (FuncDecl.get_name d) in
-             let v = StringMap.find symbol_name ctx.ctx_z3vars in
-             Format.fprintf fmt "%s %s : %s"
-               (Cli.with_style [ANSITerminal.blue] "%s" "-->")
-               (Cli.with_style [ANSITerminal.yellow] "%s" (Bindlib.name_of v))
-               (print_z3model_expr ctx (Var.Map.find v ctx.ctx_var) e)
+             match StringMap.find_opt symbol_name ctx.ctx_z3vars with
+             | None -> ()
+             | Some v ->
+               Format.fprintf fmt "%s %s : %s\n"
+                 (Cli.with_style [ANSITerminal.blue] "%s" "-->")
+                 (Cli.with_style [ANSITerminal.yellow] "%s" (Bindlib.name_of v))
+                 (print_z3model_expr ctx (Var.Map.find v ctx.ctx_var) e))
          else
            (* Declaration d is a function *)
            match Model.get_func_interp model d with

--- a/tests/test_proof/bad/enums-nonbool-empty.catala_en
+++ b/tests/test_proof/bad/enums-nonbool-empty.catala_en
@@ -1,0 +1,24 @@
+## Test
+
+```catala
+declaration enumeration T:
+   -- C content boolean
+   -- D content integer
+
+declaration enumeration S:
+    -- A content integer
+    -- B content T
+
+declaration scope A:
+  context x content integer
+  context y content S
+
+scope A:
+  definition y equals B content (D content 1)
+  definition x under condition (match y with pattern -- A of a: 1 -- B of b: 2) > 1 consequence equals 0
+  definition x under condition match y with pattern -- A of a: a < 0 -- B of b: false consequence equals 1
+```
+
+```catala-test {id="Proof"}
+catala Proof --disable_counterexamples
+```

--- a/tests/test_proof/bad/enums-nonbool-overlap.catala_en
+++ b/tests/test_proof/bad/enums-nonbool-overlap.catala_en
@@ -1,0 +1,24 @@
+## Test
+
+```catala
+declaration enumeration T:
+   -- C content boolean
+   -- D content integer
+
+declaration enumeration S:
+    -- A content integer
+    -- B content T
+
+declaration scope A:
+  context x content integer
+  context y content S
+
+scope A:
+  definition y equals B content (D content 1)
+  definition x under condition (match y with pattern -- A of a: 1 -- B of b: 2) < 2 consequence equals 0
+  definition x under condition match y with pattern -- A of a: a < 0 -- B of b: true consequence equals 1
+```
+
+```catala-test {id="Proof"}
+catala Proof --disable_counterexamples
+```

--- a/tests/test_proof/bad/output/enums-nonbool-empty.catala_en.Proof
+++ b/tests/test_proof/bad/output/enums-nonbool-empty.catala_en.Proof
@@ -1,0 +1,7 @@
+[ERROR] [A.x] This variable might return an empty error:
+  --> tests/test_proof/bad/enums-nonbool-empty.catala_en
+   |
+13 |   context x content integer
+   |           ^
+   + Test
+Counterexample generation is disabled so none was generated.

--- a/tests/test_proof/bad/output/enums-nonbool-overlap.catala_en.Proof
+++ b/tests/test_proof/bad/output/enums-nonbool-overlap.catala_en.Proof
@@ -1,0 +1,7 @@
+[ERROR] [A.x] At least two exceptions overlap for this variable:
+  --> tests/test_proof/bad/enums-nonbool-overlap.catala_en
+   |
+13 |   context x content integer
+   |           ^
+   + Test
+Counterexample generation is disabled so none was generated.

--- a/tests/test_proof/good/enums-nonbool.catala_en
+++ b/tests/test_proof/good/enums-nonbool.catala_en
@@ -1,0 +1,24 @@
+## Test
+
+```catala
+declaration enumeration T:
+   -- C content boolean
+   -- D content integer
+
+declaration enumeration S:
+    -- A content integer
+    -- B content T
+
+declaration scope A:
+  context x content integer
+  context y content S
+
+scope A:
+  definition y equals B content (D content 1)
+  definition x under condition (match y with pattern -- A of a: 1 -- B of b: 2) < 2 consequence equals 0
+  definition x under condition match y with pattern -- A of a: false -- B of b: true consequence equals 1
+```
+
+```catala-test {id="Proof"}
+catala Proof --disable_counterexamples
+```

--- a/tests/test_proof/good/output/enums-nonbool.catala_en.Proof
+++ b/tests/test_proof/good/output/enums-nonbool.catala_en.Proof
@@ -1,0 +1,1 @@
+[RESULT] No errors found during the proof mode run.

